extra/import_data.py

```diff
@@ -1,197 +1,158 @@
-
+logger.info("File upload script 'import_data.py' started")
 
-def argparser():
-    # function to enable commandline help functions. Makes use of the argparse installation.
-    # also makes sure that required parameters are used, with basic file extension checks
-
-    # Override of the error function in ArgumentParser from argparse
-    # To display the original error message + help page in the terminal, when the wrong or no arguments are given
-    class DefaultHelpParser(argparse.ArgumentParser):
-        def error(self, message):
-            sys.stderr.write('\nerror: %s\n' % message)
-            self.print_help()
-            logger.debug('error: %s' % message)
-            sys.exit(1)  # exit program
-
-    # Override of adding an extra action class in Action from argparse
-    # To add an extra action that can check for file extensions and display the help page if wrong file extension
-    def CheckExt(choices):
-        class Act(argparse.Action):
-            def __call__(self, parser, namespace, fname, option_string=None):
-                for f in fname:
-                    ext = os.path.splitext(f)[1][1:]
-                    if ext not in choices:
-                        option_string = '({})'.format(option_string) if option_string else ''
-                        parser.error("file doesn't end with {}{}".format(choices, option_string))  # display error message
-                    else:
-                        setattr(namespace, self.dest, fname)
-        return Act
-
-    # required arguments and optional arguments for the terminal. A help page is automatically generated.
-    parser = DefaultHelpParser(
-        description=(
-            "VUSualizer import data:\n"
-            "Imports data from 'O:\AMG_bijlagen_WES_uitslagen' after transferring this data into\n"
-            "the Dwergeik UMCU server. An additional script will notice new files in this folder\n"
-            "and use these as a parameter for this script. This will update the local MongoDB."
-        ),
-        formatter_class=argparse.RawTextHelpFormatter)
-    parser.add_argument(
-        "-v",
-        "--version",
-        action='version',
-        version='%(prog)s 1.0')
-    # puts following arguments under 'required', rather than the default 'optional'
-    required_argument = parser.add_argument_group('required arguments')
-    # add other file-extensions if needed: for example: CheckExt({'xlsx', 'xls})
-    required_argument.add_argument(
-        "-f",
-        "--file",
-        metavar='',
-        required=True,
-        nargs='+',
-        action=CheckExt({'xlsx'}),
-        help=(".xlsx input, usage: -f /path/to/files/XXX.xlsx (or if multiple files, "
-              "seperate with space: -f /path/to/files/XXX.xlsx /path/to/files/XXX2.xlsx'"))
-
-    args = parser.parse_args()
-    return args.file
+# start_time of script functions
+start_time = datetime.now(timezone.utc).isoformat()
```

Comment:
Also move to `main` function.

Reply:
if moved to main, it cannot be read outside of that function. I either need to return it or make it global. I thought by putting it outside the main, I can use this anywhere, including the end_time calculation.
I also want to make it as soon as possble (before connecting to Alissa), because in theory in those seconds, another analysis could be uploaded and therefore missed entirely.

Reply:
It is only used in the main function? And otherwise you should add it as a function parameter. Using global variables is typically considered bad practices, it also makes reusing functions harder / not possible.

Reply:
will make the main according to best practice and not have global variables